src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/emitter/Emitter.kt

```diff
@@ -626,7 +626,7 @@ class Emitter(
                     writeBlockComment()
                     if (event is ScalarEvent) {
                         analysis = analyzeScalar((event as ScalarEvent).value)
-                        if (!analysis!!.empty) {
+                        if (!analysis!!.empty || ((event as ScalarEvent).tag ?: "null") == Tag.STR.value) {
```

Comment:
Hmm, maybe we can get rid of `?: "null"` since it looks like it was a workaround for how Java works with Optionals.